data/json/recipes/chem/fuel.json

```diff
@@ -88,5 +88,29 @@
     ],
     "//": "Making kerosene from (tainted) fat directly is less efficient than using lard/tallow, but allows processing of spoilt fats. See #29770.  Needs 10% of volume of the fats in washing soda.  I've reduced the yield percentages from what was acheived in this laboratory based article Performance of Thermochemical Conversion of Fat, Oils, and Grease into Kerosene-like Fuel in Different Production Scales.",
     "flags": [ "ALLOW_ROTTEN" ]
+  },
+  {
+    "result": "crude_lamp_oil",
+    "type": "recipe",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_CHEM",
+    "subcategory": "CSC_CHEM_FUEL",
+    "skill_used": "chemistry",
+    "difficulty": 2,
+    "time": "5 m",
+    "batch_time_factors": [ 80, 5 ],
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 2 } ],
+    "tools": [ [ [ "surface_heat", 10, "LIST" ] ] ],
+    "components": [
+      [
+        [ "edible_tallow_lard", 2, "LIST" ],
+        [ "tallow_tainted", 2 ],
+        [ "cooking_oil", 16 ],
+        [ "cooking_oil2", 16 ],
+        [ "any_fat", 2, "LIST" ]
```

Comment:
As far as I know, animal fats and oils should be made into candles to be used for lighting, rather than burned in oil lamps as-is. The two exceptions I'm aware of are ghee (basically clarified butter) and fish oil, though the latter doesn't exist in game at the moment.

In fact I'm not even sure what exactly "animal cooking oil" represents, but this isn't directly related to this PR.

Reply:
As far as I know, animal fats were used by different cultures for the use of lighting, although especially the romans heavily preferred olive oil (perhaps because the animal fat lamps were smelly, but also olive oil was ubiquitous). I could definitely see gameplay reasons to limit lamps to veggie oil, to have it be more different from candles. But I think there is enough evidence to suggest animal fats and blubber were used at least when nothing better was available. I'll add some more scientific sources above for the usage of "grease" lamps.
But yeah, animal cooking oil *is* weird in its own right, I totally agree. Its effecitvely just liquid fat, right? Which should be solid at room temperature?

Maybe it would be sensible to reduce the crafting time, since actually not much is being done with the fat/oil? I was thinking about the process of heating up the animal fat to make it usable, but actually that doesn't seem to be necessary as far as I understand.

Reply:
"Animal cooking oil" was discussed in an issue IIRC and no one really knew what it was supposed to be